src/ip_notifier.cpp

```diff
@@ -75,7 +76,12 @@ struct ip_change_notifier_impl final : ip_change_notifier
 	explicit ip_change_notifier_impl(io_service& ios)
 		: m_socket(ios
 			, netlink::endpoint(netlink(NETLINK_ROUTE), RTMGRP_IPV4_IFADDR | RTMGRP_IPV6_IFADDR))
-	{}
+	{
+		// Linux can generate ENOBUFS if the socket's buffers are full
+		// don't treat it as an error
```

Comment:
receiving this message indicates that there was an event that got dropped, right?
I would also imagine that once the socket has been drained, you wouldn't get these anymore, until there's a spike of events overrunning the buffer again.

Are you sure getting these events isn't an indication of the event handler failing to drain the socket?
does the handler currently keep reading from the socket until it returns ``EAGAIN``, or does it only read a single message?

Reply:
well, I'm not sure to be honest, I will need some help from someone that has more knowledge with netlink.

Since I don't know for sure if this is a broadcast or unicast listener, then I'm not sure of what buffer we are dealing with.

It seems to me like this is at least an improvement, by avoiding possible CPU spinning under stress or faulty netlink behavior.

Reply:
The current handler only reads one message at a time, but the handle immediately issues a new receive request so the socket's buffer _should_ eventually be drained.

@aldenml Once the ip notifier starts getting `ENOBUFS` does it ever stop?

Reply:
at least not before making the application crash (and the phone gets warm). It happened a handful of times, but I can't find a way to reproduce it, probably some special condition in the phone.